homeassistant/components/smartthings/fan.py

```diff
@@ -92,6 +142,33 @@ def is_on(self) -> bool:
         return self._device.status.switch
 
     @property
-    def percentage(self) -> int:
+    def percentage(self) -> int | None:
         """Return the current speed percentage."""
-        return ranged_value_to_percentage(SPEED_RANGE, self._device.status.fan_speed)
+        if FanEntityFeature.SET_SPEED in self._attr_supported_features:
```

Comment:
I suppose this also don't need to be guarded as percentage would only be used if support for set_speed exist?